synapseclient/core/download/download_functions.py

```diff
@@ -161,6 +170,105 @@ async def download_file_entity(
     entity.cacheDir = os.path.dirname(download_path)
 
 
+async def download_file_entity_model(
+    *,
+    download_location: str,
+    file: "File",
+    if_collision: str,
+    submission: str,
+    synapse_client: Optional["Synapse"] = None,
+) -> None:
+    """
+    Download file entity
+
+    Arguments:
+        download_location: The download location
+        entity:           The File object
+        if_collision:      Determines how to handle file collisions.
+                            May be
+
+            - `overwrite.local`
+            - `keep.local`
+            - `keep.both`
+
+        submission:       Access associated files through a submission rather than through an entity.
+    """
+    from synapseclient import Synapse
+
+    client = Synapse.get_client(synapse_client=synapse_client)
+    # set the initial local state
+    file.path = None
+
+    # check to see if an UNMODIFIED version of the file (since it was last downloaded) already exists
+    # this location could be either in .synapseCache or a user specified location to which the user previously
+    # downloaded the file
+    cached_file_path = client.cache.get(
+        file_handle_id=file.data_file_handle_id, path=download_location
+    )
+
+    # location in .synapseCache where the file would be corresponding to its FileHandleId
+    synapse_cache_location = client.cache.get_cache_dir(
+        file_handle_id=file.data_file_handle_id
+    )
+
+    file_name = (
+        file.file_handle.file_name
+        if cached_file_path is None
+        else os.path.basename(cached_file_path)
+    )
+
+    # Decide the best download location for the file
+    if download_location is not None:
+        # Make sure the specified download location is a fully resolved directory
+        download_location = ensure_download_location_is_directory(download_location)
+    elif cached_file_path is not None:
+        # file already cached so use that as the download location
+        download_location = os.path.dirname(cached_file_path)
+    else:
+        # file not cached and no user-specified location so default to .synapseCache
```

Comment:
nit: would be nice to have a warning output message notifying the user of this behavior

Reply:
There is actually an info level message that is printed in `download_by_file_handle` that will specify the directory where the file was downloaded to:
`syn.logger.info(f"Downloaded {synapse_id} to {downloaded_path}")`

So it'll point you to the `.synapseCache` directory in this case